src/kvstore/kvstore_dist.h

```diff
@@ -144,9 +144,9 @@ class KVStoreDist : public KVStoreLocal {
     if (get_rank() == 0) {
       Push_(keys, values, 0, false);
       // wait until the push is finished
-      for (const auto& v : values) {
```

Comment:
Good catch! The comment `// wait until the push is finished` is not correct, since on GPU `v.WaitToWrite` returns as soon as v is copied/assigned to send_buf...